service/history/execution/mutable_state_decision_task_manager.go

```diff
@@ -272,7 +278,47 @@ func (m *mutableStateDecisionTaskManagerImpl) AddDecisionTaskScheduleToStartTime
 	// Clear stickiness whenever decision fails
 	m.msb.ClearStickyness()
 
-	event := m.msb.hBuilder.AddDecisionTaskTimedOutEvent(scheduleEventID, 0, workflow.TimeoutTypeScheduleToStart)
+	event := m.msb.hBuilder.AddDecisionTaskTimedOutEvent(
+		scheduleEventID,
+		0,
+		workflow.TimeoutTypeScheduleToStart,
+		"",
+		"",
+		common.EmptyVersion)
+
+	if err := m.ReplicateDecisionTaskTimedOutEvent(workflow.TimeoutTypeScheduleToStart); err != nil {
+		return nil, err
+	}
+	return event, nil
+}
+
+func (m *mutableStateDecisionTaskManagerImpl) AddDecisionTaskResetTimeoutEvent(
+	scheduleEventID int64,
+	baseRunID string,
+	newRunID string,
+	forkEventVersion int64,
+) (*workflow.HistoryEvent, error) {
+	opTag := tag.WorkflowActionDecisionTaskTimedOut
+	if m.msb.executionInfo.DecisionScheduleID != scheduleEventID || m.msb.executionInfo.DecisionStartedID > 0 {
+		m.msb.logger.Warn(mutableStateInvalidHistoryActionMsg, opTag,
+			tag.WorkflowEventID(m.msb.GetNextEventID()),
+			tag.ErrorTypeInvalidHistoryAction,
+			tag.WorkflowScheduleID(scheduleEventID),
+		)
+		return nil, m.msb.createInternalServerError(opTag)
+	}
+
+	// Clear stickiness whenever decision fails
+	m.msb.ClearStickyness()
+
+	event := m.msb.hBuilder.AddDecisionTaskTimedOutEvent(
+		scheduleEventID,
+		0,
+		workflow.TimeoutTypeScheduleToStart,
```

Comment:
Should it be TimeoutTypeReset?

Reply:
Added the reason field for reset

Reply:
I actually meant cause == RESET. it's strongly typed and can be safely used for programming.

Reply:
updated